### PR TITLE
Add conflict for MOVING_NEST option with GNU Fortran, use mcmodel=small on macOS arm64 (also contains #2017)

### LINKED ISSUE #2151: Add a recursive git submodule status to logs
## Description

Since UFSWM PR's heavily rely on components inside of components like FV3 for example it would be beneficial to track those extra subcomponents in the RT logs

## Solution

add `--recursive` to `git submodule status` in `rt.sh`
`git submodule status --recursive`

### LINKED ISSUE #2155: gfortran option `-mcmodel=medium` not working on macOS with gfortran@12.3.0
## Description
The gfortran compiler option was recently added as a default option, although I think it should have only been added for Hercules/GNU. On my macOS with gfortran@12.3.0, this doesn't work:
```
[  0%] Building Fortran object FV3/ccpp/physics/CMakeFiles/ccpp_physics.dir/physics/hooks/machine.F.o
cd /Users/heinzell/work-cs/ufs-bundle/build-atm/ufs-weather-model/src/ufs-weather-model-build/FV3/ccpp/physics && /Users/heinzell/prod/spack-stack-1.5.1/envs/unified-env/install/apple-clang/13.1.6/openmpi-4.1.5-zhlaupe/bin/mpif90 -DDEBUG -DFV3 -DINTERNAL_FILE_NML -DMOIST_CAPPA -DMPI -DNEMS_GSM -DUSE_COND -I/Users/heinzell/work-cs/ufs-bundle/build-atm/ufs-weather-model/src/ufs-weather-model-build/FV3/ccpp/physics -I/Users/heinzell/prod/spack-stack-1.5.1/envs/unified-env/install/apple-clang/13.1.6/w3emc-2.10.0-fvjfxi2/include_d -I/Users/heinzell/prod/spack-stack-1.5.1/envs/unified-env/install/apple-clang/13.1.6/bacio-2.4.1-pksx3i2/include_4 -I/Users/heinzell/prod/spack-stack-1.5.1/envs/unified-env/install/apple-clang/13.1.6/sp-2.3.3-sxl3wtc/include_d -I/Users/heinzell/prod/spack-stack-1.5.1/envs/unified-env/install/apple-clang/13.1.6/netcdf-fortran-4.6.0-qkzzwrn/include -I/Users/heinzell/prod/spack-stack-1.5.1/envs/unified-env/install/apple-clang/13.1.6/netcdf-c-4.9.2-3kb6jx2/include -fPIC -ggdb -fbacktrace -cpp -fcray-pointer -ffree-line-length-none -fno-range-check -fallow-argument-mismatch -fallow-invalid-boz -fdefault-real-8 -fdefault-double-8 -g -O0 -fno-unsafe-math-optimizations -frounding-math -fsignaling-nans -ffpe-trap=invalid,zero,overflow -fbounds-check -mcmodel=medium -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk  -fdefault-real-8 -fdefault-double-8 -fopenmp -c /Users/heinzell/work-cs/ufs-bundle/ufs-bundle/ufs-weather-model/FV3/ccpp/physics/physics/hooks/machine.F -o CMakeFiles/ccpp_physics.dir/physics/hooks/machine.F.o
gfortran: error: unrecognized argument in option '-mcmodel=medium'
gfortran: note: valid arguments to '-mcmodel=' are: large small tiny
make[5]: *** [FV3/ccpp/physics/CMakeFiles/ccpp_physics.dir/physics/hooks/machine.F.o] Error 1
make[4]: *** [FV3/ccpp/physics/CMakeFiles/ccpp_physics.dir/all] Error 2
make[3]: *** [all] Error 2
make[2]: *** [ufs-weather-model/src/ufs-weather-model-stamp/ufs-weather-model-build] Error 2
make[1]: *** [CMakeFiles/ufs-weather-model.dir/all] Error 2
make: *** [all] Error 2
```

### To Reproduce:
See above

## Additional context
n/a

### LINKED ISSUE #2159: Moving nest application - missing dependencies?
## Description
It seems the HAFS configuration w/ moving nest uses code that only works with Intel?

On MacOS with apple-clang@13.1.6, gfortran@12.3.0, spack-stack@1.6.0 (same for 1.5.1):
```
> cmake -DCMAKE_EXE_LINKER_FLAGS="${llvm_openmp_ROOT}/lib/libomp.dylib" -DAPP=HAFSW -DMOVING_NEST=ON -DCCPP_SUITES=FV3_HAFS_v1_gfdlmp_tedmf,FV3_HAFS_v1_gfdlmp_tedmf_nonsst,FV3_HAFS_v1_thompson_tedmf_gfdlsf,FV3_global_nest_v1 -D32BIT=ON ../ufs-weather-model-contiguous-arrays-USE-IT 2>&1 | tee log.cmake
...
> make VERBOSE=1 2>&1 | tee log.make
...
[ 74%] Building Fortran object FV3/CMakeFiles/fv3atm.dir/moving_nest/fv_moving_nest_utils.F90.o
cd /Users/heinzell/work/ufs-weather-model/build_moving_nest/FV3 && /opt/homebrew/bin/gfortran-12 -DENABLE_QUAD_PRECISION -DGFS_PHYS -DGFS_TYPES -DINLINE_POST -DINTERNAL_FILE_NML -DMOIST_CAPPA -DMOVING_NEST -DNO_PARALLEL_NETCDF -DOVERLOAD_R4 -DOVERLOAD_R8 -DUSE_COND -DUSE_GFSL63 -Duse_WRTCOMP -I/Users/heinzell/work/ufs-weather-model/build_moving_nest/FV3/atmos_cubed_sphere/include/fv3 -I/Users/heinzell/work/ufs-weather-model/build_moving_nest/FV3/ccpp/mod -I/Users/heinzell/work/ufs-weather-model/build_moving_nest/FV3/ccpp/framework/src -I/Users/heinzell/work/ufs-weather-model/build_moving_nest/FV3/ccpp/physics -I/Users/heinzell/work/ufs-weather-model/build_moving_nest/stochastic_physics/mod -I/Users/heinzell/work/ufs-weather-model/build_moving_nest/FV3/upp/sorc/ncep_post.fd/include -I/Users/heinzell/prod/spack-stack-1.6.0/envs/unified-env/install/apple-clang/13.1.6/fms-2023.04-hew5ac2/include_r4 -I/Users/heinzell/prod/spack-stack-1.6.0/envs/unified-env/install/apple-clang/13.1.6/netcdf-fortran-4.6.1-jd7zltz/include -I/Users/heinzell/prod/spack-stack-1.6.0/envs/unified-env/install/apple-clang/13.1.6/netcdf-c-4.9.2-wirjmtt/include -I/Users/heinzell/prod/spack-stack-1.6.0/envs/unified-env/install/apple-clang/13.1.6/openmpi-4.1.6-doim432/include -I/Users/heinzell/prod/spack-stack-1.6.0/envs/unified-env/install/apple-clang/13.1.6/openmpi-4.1.6-doim432/lib -I/Users/heinzell/prod/spack-stack-1.6.0/envs/unified-env/install/apple-clang/13.1.6/w3emc-2.10.0-5ahp4cu/include_d -I/Users/heinzell/prod/spack-stack-1.6.0/envs/unified-env/install/apple-clang/13.1.6/bacio-2.4.1-rwstqew/include_4 -I/Users/heinzell/prod/spack-stack-1.6.0/envs/unified-env/install/apple-clang/13.1.6/esmf-8.5.0-axnwouh/include -I/Users/heinzell/prod/spack-stack-1.6.0/envs/unified-env/install/apple-clang/13.1.6/parallelio-2.5.10-w4vaq4m/include -I/Users/heinzell/prod/spack-stack-1.6.0/envs/unified-env/install/apple-clang/13.1.6/crtm-2.4.0.1-pivdurm/include -I/Users/heinzell/prod/spack-stack-1.6.0/envs/unified-env/install/apple-clang/13.1.6/g2-3.4.5-tudq2li/include_4 -I/Users/heinzell/prod/spack-stack-1.6.0/envs/unified-env/install/apple-clang/13.1.6/libpng-1.6.37-gxbp73o/include -I/Users/heinzell/prod/spack-stack-1.6.0/envs/unified-env/install/apple-clang/13.1.6/zlib-1.2.13-uraowup/include -I/Users/heinzell/prod/spack-stack-1.6.0/envs/unified-env/install/apple-clang/13.1.6/g2tmpl-1.10.2-cadrhw2/include -ggdb -fbacktrace -cpp -fcray-pointer -ffree-line-length-none -fno-range-check -fallow-argument-mismatch -fallow-invalid-boz -O2 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -Jmod -fopenmp -c /Users/heinzell/work/ufs-weather-model/ufs-weather-model-contiguous-arrays-USE-IT/FV3/moving_nest/fv_moving_nest_utils.F90 -o CMakeFiles/fv3atm.dir/moving_nest/fv_moving_nest_utils.F90.o
/Users/heinzell/work/ufs-weather-model/ufs-weather-model-contiguous-arrays-USE-IT/FV3/moving_nest/fv_moving_nest_utils.F90:85:7:

   85 |   use ifport,            only: getcwd
      |       1
Fatal Error: Cannot open module file 'ifport.mod' for reading at (1): No such file or directory
compilation terminated.
make[2]: *** [FV3/CMakeFiles/fv3atm.dir/moving_nest/fv_moving_nest_utils.F90.o] Error 1
make[1]: *** [FV3/CMakeFiles/fv3atm.dir/all] Error 2
make: *** [all] Error 2
```

https://ahamodel.uib.no/intel/GUID-01E8863C-CAD1-4345-B674-F75262B7152A.html 

If this option only works with Intel, then there should be a conflict right away in the cmake call if the compiler is not Intel?

### PR DESCRIPTION
**Pull request by climbfuji**
## Commit Queue Requirements:

- [x] Fill out all sections of this template.
- [x] All sub component pull requests have been reviewed by their code managers.
- [x] Run the full Intel+GNU RT suite (compared to current baselines) on either Hera/Derecho/Hercules
- [ ] ~~Commit 'test_changes.list' from previous step~~ n/a - no changes to baseline, file wasn't produced during RT run

---
## Description:

1. This PR contains all the changes from https://github.com/ufs-community/ufs-weather-model/pull/2017
2. Fix a compiler error on macOS arm64, where `gfortran` doesn't understand `-mcmodel=medium`.
3. Throw an error when trying to compile the `MOVING_NEST` option with GNU (only supported for Intel)
3. Submodule pointer update for fv3atm and ccpp-framework for changes listed in the associated PRs below

### Testing

Aside from regression testing:
1. I tested that the MOVING_NEST build errors out immediately with GNU Fortran:
```
CMake Error at cmake/GNU.cmake:3 (message):
  Option MOVING_NEST not compatible with GNU}
Call Stack (most recent call first):
  CMakeLists.txt:137 (include)
```
4. I tested that the `atm_dyn32_phy32_debug` compile option compiles successfully on my macOS arm64

### Commit Message:

```
* UFSWM - "Add conflict for MOVING_NEST option with GNU Fortran, use mcmodel=small on macOS arm64"
  * AQM - 
  * CDEPS - 
  * CICE - 
  * CMEPS - 
  * CMakeModules - 
  * FV3 - "Bug fix for unallocated array sfcprop%wetness in GFS_diagnostics.F90"
    * ccpp-physics - 
    * ccpp-framework - "Bug fix required to use chunked arrays and retain backward compatibility with blocked data structures in ccpp_prebuild.py"
    * atmos_cubed_sphere - 
  * GOCART - 
  * HYCOM - 
  * MOM6 - 
  * NOAHMP - 
  * WW3 - 
  * stochastic_physics - 
```

### Priority:
* Normal.

### UFSWM:

Resolves https://github.com/ufs-community/ufs-weather-model/issues/2159
Resolves https://github.com/ufs-community/ufs-weather-model/issues/2155
Resolves https://github.com/ufs-community/ufs-weather-model/issues/2151

### Sub component Pull Requests:
* AQM:
* CDEPS:
* CICE:
* CMEPS:
* CMakeModules:
* FV3: https://github.com/NOAA-EMC/fv3atm/pull/792
  * ccpp-physics:
  * ccpp-framework: https://github.com/NCAR/ccpp-framework/pull/538
  * atmos_cubed_sphere:
* GOCART:
* HYCOM:
* MOM6:
* NOAHMP:
* WW3:
* stochastic_physics:
* None

### UFSWM Blocking Dependencies:

This is part of a set of (independent) PRs:
- waiting on https://github.com/NCAR/ccpp-framework/pull/538
- waiting on https://github.com/NOAA-EMC/fv3atm/pull/792
- https://github.com/ufs-community/ufs-weather-model/pull/2164

### Regression Test Changes (Please commit test_changes.list):

No Baseline Changes.

### Input data Changes:

None

### Library Changes/Upgrades:

None
  
---

## Testing Log:
- RDHPCS
  - [x] Hera
  - [x] Orion
  - [x] Hercules
  - [x] Jet
  - [x] Gaea
  - [x] Derecho
- WCOSS2
  - [x] Dogwood/Cactus
  - [ ] Acorn
- [x] CI
- [ ] opnReqTest (complete task if unnecessary)

### CODE REVIEW

**Review by BrianCurtis-NOAA — DISMISSED** [review]
@climbfuji Thanks!

**Comment by zach1221** [comment]
Hello, @climbfuji . Would you be comfortable syncing up this PR, and combining in the changes from PR #2017 ?

**Comment by climbfuji** [comment]
Sure, let me do that.

**Comment by climbfuji** [comment]
Done. This PR is ready for review and testing.

**Comment by jkbk2004** [comment]
@BinLiu-NOAA @JessicaMeixner-NOAA @jiandewang @MatthewPyle-NOAA @grantfirl @junwang-noaa With this PR, we are moving to Rocky8 nodes on Jet: https://github.com/ufs-community/ufs-weather-model/pull/2164/files#diff-ba51bc885d1d3ab974294b540115871708f266e15ec6d7dee5df104000fd24ceL33. I like to check if any experiment is going on Centos nodes on jet. Or if we need keep centos nodes, we can revert jet modulefile update. Note that ecflow on centos nodes may not be functional as well.

**Comment by BinLiu-NOAA** [comment]
@jkbk2004 Fine from HAFS end. We are actually waiting for and looking forward to this Rocky8 upgrade/transition. Thanks!

**Comment by zach1221** [comment]
@jkbk2004 @BrianCurtis-NOAA are you both good with the current Jet code changes?

**Comment by BrianCurtis-NOAA** [comment]
@zach1221 Code looks OK, fix the indentation to match please.

File: tests/rt.sh [hunk 1]
```diff
@@ -752,6 +752,13 @@ elif [[ $MACHINE_ID = hercules ]]; then
 
 elif [[ $MACHINE_ID = jet ]]; then
 
+  echo "=======Running on $(lsb_release -is)======="
+CurJetOS="$(lsb_release -is)"
```
Comment (BrianCurtis-NOAA) [inline tests/rt.sh]: I take that back. You'll want something more like:
```
  CurJetOS=$(lsb_release -is)
  if [[ ${CurJetOS} == "CentOS" ]]; then
```

Comment (zach1221) [reply tests/rt.sh]: Ok, done. @BrianCurtis-NOAA

**Comment by zach1221** [comment]
@BrianCurtis-NOAA We can add issue 2151 in here.

**Comment by BrianCurtis-NOAA** [comment]
@zach1221 Have we ran on all machines with the updated --recursive option for the git submodule status? I think some older git packages do not support that option correctly, but i'm not sure which machines might be using older git versions.

**Comment by zach1221** [comment]
> @zach1221 Have we ran on all machines with the updated --recursive option for the git submodule status? I think some older git packages do not support that option correctly, but i'm not sure which machines might be using older git versions.

@BrianCurtis-NOAA I haven't run on the other machines yet, but I'll keep an eye out for issues.

**Comment by BrianCurtis-NOAA** [comment]
Acorn: We are still working on addressing the build/run issues there. Can skip.

**Comment by zach1221** [comment]
Ok, testing is complete here. We can begin merge process starting with the ccpp-framework sub-pr.

**Comment by zach1221** [comment]
@climbfuji fv3atm is now merged as well.
https://github.com/NOAA-EMC/fv3atm/commit/062f7958ccd9bd97b8737432f75b06ec734777a7

**Comment by climbfuji** [comment]
This should be ready for merging, please check

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Review by jkbk2004 — APPROVED** [review]

**Pull request merged**